flutter/lib/src/navigation/sentry_navigation_observer.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/widgets.dart';
+import 'package:sentry/sentry.dart';
+
+import 'navigation_breadcrumb.dart';
+
+/// This is a navigation observer to record navigational breadcrumbs.
+/// For now it only records navigation events and no gestures.
+///
+/// Routes can always be null and their settings can also always be null.
+/// For example, if the application starts, there is no previous route.
+/// The RouteSettings are null if a developer has not specified any
+/// RouteSettings.
+///
+/// SentryNavigationObserver must be added to the navigation observer of
+/// your used app. This is an example for [MaterialApp](https://api.flutter.dev/flutter/material/MaterialApp/navigatorObservers.html),
+/// but the integration for [CupertinoApp](https://api.flutter.dev/flutter/cupertino/CupertinoApp/navigatorObservers.html)
+/// and [WidgetsApp](https://api.flutter.dev/flutter/widgets/WidgetsApp/navigatorObservers.html) is the same.
+/// ´´´dart
+/// MaterialApp(
+///   navigatorObservers: [
+///     SentryNavigatorObserver(),
+///   ],
+///   // other parameter ...
+/// )
+/// ´´´
+///
+/// See also:
+///   - https://api.flutter.dev/flutter/widgets/RouteObserver-class.html
+class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
+  factory SentryNavigatorObserver({Hub hub}) {
+    if (hub != null) {
+      return SentryNavigatorObserver._(hub);
+    }
+    return SentryNavigatorObserver._(Sentry.currentHub);
+  }
```

Comment:
instead of using the `Sentry.currentHub` I'd rather use the `HubAdapter` which is basically redirecting to `Sentry.currentHub`, so we don't have a static class reference and still, unit testable.